pulsar/consumer_partition.go

```diff
@@ -596,7 +596,7 @@ func (pc *partitionConsumer) MessageReceived(response *pb.CommandMessage, header
 		msgID := newTrackingMessageID(
 			int64(pbMsgID.GetLedgerId()),
 			int64(pbMsgID.GetEntryId()),
-			int32(i),
+			pbMsgID.GetBatchIndex(),
```

Comment:
Could there be compatibility issues if users upgrade? Is this what the java client does too?

Reply:
Yes, this may break compatibility. The Java SDK side obtains the Batch Idx information from the Broker instead of maintaining the index information locally.

Therefore, if the user uses the Java SDK on the producer side and the Go SDK on the consumer side, there is no way to match the attributes of the message ID.

This effect is relatively small and only affects non-partitioned topic types. In most cases, users may prefer to use partitioned-topic